### PR TITLE
JS-2064 New rule S9107: vue require prop type constructor

### PR DESCRIPTION
**Pull request by guillemsarda**
Part of 

----
## Summary by Gitar

- **New Rule Implementation:**
    - Added rule `S9107` (`vue/require-prop-type-constructor`) ensuring Vue component prop types are constructors.
    - Included rule metadata, decorator, unit tests, JSON configuration, and documentation updates.

<sub>This will update automatically on new commits.</sub>

### CODE REVIEW

File: packages/analysis/src/jsts/rules/S9107/decorator.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
+// https://sonarsource.github.io/rspec/#/rspec/S9107/javascript
+
+import type { Rule } from 'eslint';
+import { generateMeta } from '../helpers/generate-meta.js';
+import { interceptReport } from '../helpers/decorators/interceptor.js';
+import * as meta from './generated-meta.js';
+
+const IDENTIFIER_PATTERN = /^[A-Z][A-Za-z0-9_$]*$/;
```
Comment (nathsou) [inline packages/analysis/src/jsts/rules/S9107/decorator.ts:24]: Preserve the safe `null` quick-fix

`IDENTIFIER_PATTERN` rejects `null`, so `type: 'null'` is now reported without the upstream fix. However, `type: null` is accepted by the upstream Vue rule (and is specified as an exception in the RSPEC: https://github.com/SonarSource/rspec/blob/8fb75c3430f746d53a16e02a49e10d68621850b6/rules/S9107/javascript/rule.adoc#exceptions) and therefore by this decorator, the decorator only changes reports it receives. This turns a valid, issue-resolving fix into no fix. Please allow this case (and add a regression test for `'null'` → `null`).

**Review by nathsou — APPROVED** [review]
(Just 1 comment really)

Image
